product-service/src/main/java/com/extremecoder/productservice/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,164 @@
+    public final ResponseEntity<Object> handleHttpMessageNotReadableError(Exception ex) {
+        log.error("{}", ex);
+        return ResponseEntity
+                .status(HttpStatus.BAD_REQUEST.value())
+                .header(CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .body(ex.getMessage());
+    }
+
+    /**
+     * Handle Resource Not Found Exception
+     *
+     * @param ex NoHandlerFoundException
+     * @return Response Entity
+     */
+    @ExceptionHandler({NoHandlerFoundException.class})
+    public ResponseEntity<Object> handleResourceNotFoundException(NoHandlerFoundException ex) {
+        // todo: modify response
+        log.error("{}", ex);
+        return ResponseEntity
+                .status(HttpStatus.NOT_FOUND)
+                .header(CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .body("");
+    }
+
+    @ExceptionHandler({HttpRequestMethodNotSupportedException.class})
+    public final ResponseEntity<Object> handleMethodNotAllowedException(Exception ex) {
+        // todo: modify response
+        log.error("{}", ex);
+        return ResponseEntity
+                .status(HttpStatus.METHOD_NOT_ALLOWED.value())
+                .header(CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .build();
+    }
+
+    /**
+     * Handle Required request parameter response entity.
+     *
+     * @return Response Entity
+     */
+    @ExceptionHandler({MissingServletRequestParameterException.class})
+    public final ResponseEntity<Object> handleMissingServletRequestParameterException(MissingServletRequestParameterException ex) {
+        // todo: modify response
+        log.error("{}", ex);
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    }
+
+    /**
+     * Handle Illegal Argument Exception
+     *
+     * @param ex IllegalArgumentException
+     * @return Response Entity
+     */
+    @ExceptionHandler({IllegalArgumentException.class})
+    public ResponseEntity<?> handleIllegalArgumentException(Throwable ex) {
+        // todo: modify response
+        log.error("{}", ex);
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    }
+
+    /**
+     * Handle internal error.
+     *
+     * @param ex of type {@link RuntimeException}
+     * @return the response entity
+     */
+    @ExceptionHandler({RuntimeException.class})
+    public ResponseEntity<Object> handleInternalException(Throwable ex) {
+        // todo: modify response
+        log.error("{}", ex);
+        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+    }
+
+    /**
+     * Handle Bind Exception
+     * -->
+     *
+     * @param ex BindException
+     * @return the response entity
+     */
+    @ExceptionHandler({BindException.class})
+    public ResponseEntity<Object> handleBindException(BindException ex) {
+        // todo: modify response
+        log.error(ex.getMessage());
+
+        Map<String, Object> body = new HashMap<>();
```

Comment:
Instead of creating new variable we can achieve the same thing with `Map.of()` method which will save us an extra variable creation.

Reply:
done